docs/task-cluster.md

```diff
@@ -97,6 +97,20 @@ tasks:
             stage: merge-corpus
 ```
 
+## Running only later parts of the pipeline
+
+When hacking on later parts of the pipeline it can often be useful to re-use earlier runs of the pipeline, even if those runs were done with different training parameters. To do this, we must bypass the usual caching mechanisms of Taskgraph, and force it to replace earlier tasks with ones we provide. To do this, you can run a training action as usual, but also provide `start-stage` and `previous_group_ids` parameters. For example:
+
+```
+start-stage: train-student
+target-stage: all
+previous_group_ids: ["SsGpi3TGShaDT-h93fHL-g"]
+```
+
+...will run `train-student` and all tasks _after_ it. All tasks upstream of `train-student` will be replaced with the tasks of the same name from the `SsGpi3TGShaDT-h93fHL-g` task group. (If there are required tasks that do not exist in any of the provided groups, they will be created - so you must ensure that any tasks you want to skip exist in one of the previous groups.)
```

Comment:
> so you must ensure that any tasks you want to skip exist in one of the previous groups

I think this might become problematic because sometimes we restart an experiment many times and some cached tasks might be from older experiments. I think the code in the train action should just traverse recursively all the cached tasks and add them too. Then we can specify just one group id. Also specifying multiple group IDs will make it ambiguous if a task is duplicated in both groups.

Reply:
That's fair - I don't see any reason that we can't do this.

Reply:
So we'd specify some tasks that are existing, and then fallback to the cache?

Reply:
I think how this would work is we would take any tasks from the group id specified that we need, and are successful, and then walk through their dependencies for any other tasks that we need. (Essentially, we're going to do some extra work in the action task to discover additional existing tasks rather than require them to be specified.)

Reply:
I just pushed an update here that should address this. It requires that a `previous_group_id` that contains the task or tasks from the `start-stage` to exist in it - but after that it will walk back up the dependency chain to find the rest.

This prevents one from starting, eg: `train-teacher` off of a group that only ran up to `train-backwards`, but based on the description of the use case I think this is generally going to be used when you've already run to task X and then want to run it again.

@eu9ene - does this address your concerns?

(It might be possible to address this limitation in the future, but due to the limited data that action tasks have in them it's not trivial...)

This new version also depends on https://github.com/taskcluster/taskgraph/pull/418 to be merged and released.

Reply:
Oh, and you can give it a try right now by running training actions off of https://firefox-ci-tc.services.mozilla.com/tasks/groups/Ri8ENf-AReefjxNRxBESEQ.

Reply:
Great, this should work for us. Let's merge and I'll test it later.